File: frontend/spa/src/utils/clientTimeZone.js
/** IANA TZ браузера (Europe/Moscow и т.д.) для заголовка X-Client-Timezone. */
export function getBrowserIanaTimeZone() {
  try {
    return Intl.DateTimeFormat().resolvedOptions().timeZone || "";
  } catch {
    return "";
  }
}

File: src/api/client_timezone.py
"""Часовой пояс браузера клиента для контекста LLM (IANA, например Europe/Moscow)."""

from __future__ import annotations

from typing import Annotated

from fastapi import Header

ClientTimezoneIdDep = Annotated[
    str | None,
    Header(
        alias="X-Client-Timezone",
        description="IANA Time Zone Database, например Europe/Moscow (Intl.DateTimeFormat().resolvedOptions().timeZone)",
    ),
]
